app/src/main/java/com/totsp/crossword/BackgroundDownloadService.java

```diff
@@ -0,0 +1,156 @@
+package com.totsp.crossword;
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.NotificationManager;
+import android.app.job.JobInfo;
+import android.app.job.JobParameters;
+import android.app.job.JobScheduler;
+import android.app.job.JobService;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.os.AsyncTask;
+import android.os.Looper;
+import android.preference.PreferenceManager;
+import android.support.v4.content.ContextCompat;
+
+import com.totsp.crossword.net.Downloaders;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// Currently only available on API version >=21 due to use of JobScheduler.
+// It may be possible to implement this functionality using AlarmManager for lower SDK versions.
+@TargetApi(21)
+public class BackgroundDownloadService extends JobService {
+    public static final String DOWNLOAD_PENDING_PREFERENCE = "backgroundDlPending";
+
+    private static final Logger LOGGER =
+            Logger.getLogger(BackgroundDownloadService.class.getCanonicalName());
+
+    private static JobInfo getJobInfo(boolean requireUnmetered, boolean allowRoaming,
+                                      boolean requireCharging) {
+        JobInfo.Builder builder = new JobInfo.Builder(
+                JobSchedulerId.BACKGROUND_DOWNLOAD.id(),
+                new ComponentName("com.totsp.crossword.shortyz",
+                        BackgroundDownloadService.class.getName()));
+
+        builder.setPeriodic(TimeUnit.HOURS.toMillis(1))
+                .setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED)
```

Comment:
Turns out that this is a bit more complicated than planned.  It's possible to listen for timezone (and it seems DST) changes on the user's device via the TIME_CHANGED and TIMEZONE_CHANGED intents and reschedule your job/alarm accordingly.

However, we're actually trying to set an alarm in a different time zone (12AM Pacific as you mentioned).  Since not everyone changes DST at the same time (or at all, e.g. Arizona), we need to reschedule the job when the PDT<->PST transition happens to keep the alarm at the right time of day.

If we just ditch periodic jobs and reschedule after a successful completion, it might work alright since the consequences of a miss by an hour or so in either direction aren't too bad (deliver late a couple times a year, or schedule an extra job).  We'd have to make this work with handling transient download failures though.

That said, the Downloaders class will already prevent re-downloading if the puzzle exists.  I just updated the PR with a fix for a bug around notifications here (a notification still got posted even if all downloads were skipped).

Given that making this work for users in all time zones is as bit complex, and we already avoid unnecessary network access, would you be alright leaving this as is for now?